### PR TITLE
CLN: Enforce deprecation of groupby.quantile supporting bool dtype

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Ref: #53975

### CODE REVIEW

File: pandas/tests/groupby/test_numeric_only.py [hunk 1]
```diff
@@ -368,18 +368,11 @@ def test_deprecate_numeric_only_series(dtype, groupby_func, request):
...
+            method(*args)
     elif dtype is object:
-        warn = FutureWarning if groupby_func == "fillna" else None
-        warn_msg = "SeriesGroupBy.fillna is deprecated"
-        with tm.assert_produces_warning(warn, match=warn_msg):
-            result = method(*args)
-        with tm.assert_produces_warning(warn, match=warn_msg):
-            expected = expected_method(*args)
+        result = method(*args)
+        expected = expected_method(*args)
```
Comment (datapythonista) [inline pandas/tests/groupby/test_numeric_only.py:375]: Those are not related to quantile with bool, right? I'm a bit confused here.

**Review by datapythonista — APPROVED** [review]
lgtm, just some changes seem to be unrelated, but if it's something pending that you're addressing here, fine with me.

File: pandas/tests/groupby/test_numeric_only.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/tests/groupby/test_numeric_only.py:375]: Yea - this was left over from #57109; the `fillna` method no longer exists on groupby.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rhshadrach
